var/spack/repos/builtin/packages/hipsycl/package.py

```diff
@@ -73,15 +79,17 @@ def cmake_args(self):
         args = [
             "-DWITH_CPU_BACKEND:Bool=TRUE",
             # TODO: no ROCm stuff available in spack yet
-            "-DWITH_ROCM_BACKEND:Bool=FALSE",
+            "-DWITH_ROCM_BACKEND:Bool={0}".format("TRUE" if "+rocm" in spec else "FALSE"),
             "-DWITH_CUDA_BACKEND:Bool={0}".format("TRUE" if "+cuda" in spec else "FALSE"),
             # prevent hipSYCL's cmake to look for other LLVM installations
             # if the specified one isn't compatible
             "-DDISABLE_LLVM_VERSION_CHECK:Bool=TRUE",
         ]
+        if "+rocm" in spec:
+            args += [f"-DHIP_CXX_COMPILER={self.compiler.cxx}"]
```

Comment:
Without this I was getting an issue where CMake was trying to locate a header file WRT the HIP compiler, which defaults to the Spack compiler wrapper (which results in a failure)

Reply:
Good catch. Looks like AdaptiveCpp had somewhat broken CMake detection.

I tried to make a fix: https://github.com/AdaptiveCpp/AdaptiveCpp/compare/try-more-robust-cmake-rocm-search. If you have time, could you try of it helps in the spack setup? But we'll have to keep the current solution anyway for all the already released versions, so it's not critical; no way around setting `HIP_CXX_COMPILER` for now.

A bit more concerning: while setting HIP_CXX_COMPILER is necessary to find ROCm, it is definitely not necessary to use `hipcc` as the default C++ compiler for the project. At least we, when building hipSYCL/AdaptiveCpp for GROMACS, only use either mainline `clang` or `amdclang` from ROCm package; and officially, AdaptiveCpp only tests mainline `clang`. I guess it might work, but I am not sure about the interplay of different flags, since some ACpp headers and its compiler wrapper add/define the same macros `hipcc` defines, and the behavior might be interesting if there is there is a clash.